### ISSUE TITLE
accessibility: prev / next navigation landmark  does not have a unique name

### ISSUE THREAD

**Issue by gorlanova** [vuejs/vitepress#3516]
### Describe the bug

Landmarks should have a unique role or role/label/title (i.e. accessible name) combination
Currently, the nav at the bottom of the page containing "prev" and "next" links does not
It makes it hard for accessibility tools to understand the difference between this one and the main one

### Reproduction

Navigate to a doc page having "prev" or "next" navigation (or both) at the bottom
Inspect the accessibility of the nav component, it does not have an accessible name

Triggers an error in Axe Devtools

### Expected behavior

Should have an accessible name to be distinct from main navigation (ideally, main navigation should have an accessible name too)

### System Info

```Text
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M2
    Memory: 393.92 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 10.2.5 - /usr/local/bin/npm
    pnpm: 8.9.2 - /opt/homebrew/bin/pnpm
  Browsers:
    Safari: 17.2.1
  npmPackages:
    vitepress: ^1.0.0-rc.31 => 1.0.0-rc.35
```

### Additional context

<img width="921" alt="Screenshot 2024-01-29 at 15 49 37" src="https://github.com/vuejs/vitepress/assets/58045985/546c396b-a86b-4550-8d35-9d75ba4d6c0d">

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by olets** [comment]
I noticed this too. The problem is arguably a side effect of the outline nav's including both a labelled `nav` and an unlabelled `role="navigation"`. But labelling the footer nav would only be an improvement.
